src/analysis/specific_limits.lean

```diff
@@ -304,6 +304,21 @@ begin
   simpa [div_eq_mul_inv] using tendsto_pow_const_div_const_pow_of_one_lt k hr'
 end
 
+/-- If `0 ≤ r < 1`, then `n ^ k r ^ n` tends to zero for any natural `k`. -/
+lemma tendsto_pow_const_mul_const_pow_of_lt_one (k : ℕ) {r : ℝ} (hr : 0 ≤ r) (h'r : r < 1) :
```

Comment:
Any reason not to generalize?
```suggestion
/-- If `0 ≤ r < 1`, then `n ^ k r ^ n` tends to zero for any natural `k`. -/
lemma tendsto_pow_const_mul_const_pow_of_lt_one (k : ℕ) {r : ℝ} (hr : -1 < r) (h'r : r < 1) :
```

Reply:
isn't that what the original statement basically is?

Reply:
The original statement is in terms of `|r| < 1`, so it's not exactly the same. But I expect that in 99% of the applications one will have `0 ≤ r` (and this will be in the context), so the point of the lemma is precisely to be less general than it could be, but direct to apply.

Reply:
Can you capture "the point of the lemma" in the docstring? Maybe
```suggestion
/-- As a common case, if `0 ≤ r < 1` then `n ^ k r ^ n` tends to zero for any natural `k`. -/
lemma tendsto_pow_const_mul_const_pow_of_lt_one (k : ℕ) {r : ℝ} (hr : 0 ≤ r) (h'r : r < 1) :
```
or
```suggestion
/-- If `0 ≤ r < 1`, then `n ^ k r ^ n` tends to zero for any natural `k`.
This is a special case of `tendsto_pow_const_mul_const_pow_of_abs_lt_one` for convience. -/
lemma tendsto_pow_const_mul_const_pow_of_lt_one (k : ℕ) {r : ℝ} (hr : 0 ≤ r) (h'r : r < 1) :
```
or similar.

Reply:
That's a good idea, thanks! Done.